chore(css): Move to vanilla CSS

qw-in opened the pull request:
This moves us entirely to vanilla css. The idea being this will be extremely portable and easy to use in every example app. It also removes a bunch of left-over tailwindcss, ports us closer to the design system (all colors are from the DS). (-2000 LoC is way easier to maintain)

It is not a 1-1 pixel perfect port. I'll need feedback from @arcjet/design-team so I know what should be brought in line and what is ok to change. I've deliberately kept things as simple as possible. It's not perfect but it is another step in the right direction and towards code sharing.

I will use this exact stylesheet in the astro example (see initial work I'll update at https://github.com/arcjet/examples/pull/1 (soon to be public)). I decided that having the diff and screenshots here would be helpful when reviewing the PR.

Some notable differences:
- Mobile fixes (Closes #272)
- Color scheme is now _only_ based on user preference (for simplicity)
- Simplified breakpoints (notably in the nav)
- Color tweaks (in line with design system)
- Other changes like spacing

qw-in commented on styles/styles.css:
```diff
@@ -0,0 +1,590 @@
+    --theme-surface: light-dark(var(--palette-neutral-15), var(--palette-neutral-00));
+    --theme-input: light-dark(var(--palette-neutral-12), var(--palette-neutral-02));
+
+    /* TODO: Use as the override (looks like we'll need javascript to toggle an attribute in 2025...) */
+    /* color-scheme: dark; */
```
Only if we decide manual dark/light mode is a requirement

morekid commented:
Noticed a bug when resizing with the open popup menu.

https://github.com/user-attachments/assets/8a27f19e-c2e4-45e0-999a-4adedd63fc0a

morekid commented:
Styling for code (inline, block) is incosistent.

We want the primary color for block level code and secondary for inline code, these where `--foreground` and `--secondary-foreground". For instance the inline emails shouldn't inherit the muted color of the text that follows, they should stay secondary.

Also needs adjusting the text size for both inline/block and padding for block code.

Before:
<img width="525" alt="image" src="https://github.com/user-attachments/assets/79e97b3f-70b4-4db7-83b7-d7522b814a75" />

<img width="407" alt="image" src="https://github.com/user-attachments/assets/7ccbaa0d-6210-4abc-a9e0-d89be0b06d28" />

After:
<img width="416" alt="Screenshot 2025-07-01 at 16 22 25" src="https://github.com/user-attachments/assets/abaa928b-15c6-48be-aa68-d6d293730150" />

<img width="462" alt="image" src="https://github.com/user-attachments/assets/ad15dbec-d39d-4186-81c3-c4575c5cce9b" />

morekid commented:
A couple more things I found.

----

There are no active states on the menu items when you're on the relevant page:

Before:
<img width="1270" alt="image" src="https://github.com/user-attachments/assets/c386b95e-4ab0-4af1-89e7-22ee0efb1c2b" />

Now:
<img width="1210" alt="image" src="https://github.com/user-attachments/assets/a585b995-4dc5-4306-a5d0-6e26675831a0" />

----

The header items should be baseline aligned:

Before:
<img width="1276" alt="image" src="https://github.com/user-attachments/assets/fc33d1de-0ea8-4196-9961-6e97f3b45313" />


Now:
<img width="1211" alt="image" src="https://github.com/user-attachments/assets/3e4294f8-ecc3-484d-b5a3-f154c8a539c4" />

qw-in commented:
@morekid thanks for your review. Very helpful and now I have a finite list of things to fix.

> Ok, I'll move it to just kebab case class names

I've completed this, let me know if you have any naming suggestions. Say what you will about tailwind but it does avoid [one of the hardest problems in computer science](https://martinfowler.com/bliki/TwoHardThings.html) 😀 (best reference I could find)

> Noticed a bug when resizing with the open popup menu.

Yeah, I had intended to respond to your weekly update question about limitations in popover about this. It _seems_ that without javascript there is no way to pull children in an open popover out of the [Top layer](https://developer.mozilla.org/en-US/docs/Glossary/Top_layer).

I've added some JavaScript that automatically closes the popover at the correct breakpoint. Unfortunate but will need to be duplicated across all the examples

> Separators in dark mode are too light

I've updated to `--palette-neutral-05` as it is the closest palette color to the live divider color (old color #4d4955).

> There are no active states on the menu items when you're on the relevant page.

I've introduced the `NavLink` component to handle this via the `data-active` attribute.

TODO:

- [ ] The header items should be baseline aligned
- [ ] Replace the very long and ugly button text
- [ ] Wrap "Want to know more?" more nicely?
- [ ] Styling for code (inline, block) is inconsistent

morekid commented:
> I've completed this, let me know if you have any naming suggestions.

Looks good, thanks for going through the chore!

> [one of the hardest problems in computer science](https://martinfowler.com/bliki/TwoHardThings.html) 😀 

So many two hard things that I don't know what's important anymore now 😂.

> limitations in popover 

Ah bummer! Haven't looked at the details of the specific implementation but I wonder if there's a way to snap the items into place with just CSS positioning when you're in the larger viewport, by explicitly setting instead of unsetting the navigation. This would have the advantage that the popover would retain the set state.

> I've updated to `--palette-neutral-05` 
> I've introduced the `NavLink` component to handle this via the `data-active` attribute.

👍 👍

morekid commented:
@qw-in mostly resolved but there are still a couple issues that I'd rather fix before a future design pass. I've put the changes in this https://github.com/arcjet/example-nextjs/pull/327 to keep things tidy. Feel free to merge and proceed or ping me if you want me to do it.
